File: C++/isomorphic_string.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {

        if (s.length()  != t.length()){
            return false;
        }

        unordered_map<char, char> mapST;
        unordered_map<char, char> mapTT;

        for (int i=0; i< s.length(); i++){

            char schar = s[i];
            char tchar = t[i];

            if (mapST.count(schar)){
                if (mapST[schar] != tchar){
                    return false;
                }

            } else {
                mapST[schar] = tchar;

            }

            if (mapTT.count(tchar)){
                if (mapTT[tchar] != schar){
                    return false;
                }

            } else {
                mapTT[tchar] = schar;

            }



        } 

        return true;


        
    }
};

File: Python/isomorphic_string.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        mapST = {}  
        mapTS = {}  

        for sChar, tChar in zip(s, t):
            if sChar in mapST:
                if mapST[sChar] != tChar:
                    return False
            else:
                mapST[sChar] = tChar
            
            if tChar in mapTS:
                if mapTS[tChar] != sChar:
                    return False
            else:
                mapTS[tChar] = sChar

        return True
